xtask/src/main.rs

```diff
@@ -0,0 +1,607 @@
+            errors,
+        );
+        do_test(
+            || {
+                run_command::<&str>(
+                    &[
+                        "cargo",
+                        "check",
+                        "--target",
+                        &target,
+                        "-p",
+                        "defmt",
+                        "--features",
+                        "alloc",
+                    ],
+                    None,
+                    &[],
+                )
+            },
+            "cross",
+            errors,
+        );
+    }
+
+    do_test(
+        || {
+            run_command(
+                &[
+                    "cargo",
+                    "check",
+                    "--target",
+                    "thumbv6m-none-eabi",
+                    "--workspace",
+                    "--exclude",
+                    "defmt-itm",
+                    "--exclude",
+                    "firmware",
+                ],
+                Some("firmware"),
+                &[],
+            )
+        },
+        "cross",
+        errors,
+    );
+
+    do_test(
+        || {
+            run_command(
+                &[
+                    "cargo",
+                    "check",
+                    "--target",
+                    "thumbv7em-none-eabi",
+                    "--workspace",
+                ],
+                Some("firmware"),
+                &[],
+            )
+        },
+        "cross",
+        errors,
+    );
+
+    do_test(
+        || {
+            run_command(
+                &[
+                    "cargo",
+                    "check",
+                    "--target",
+                    "thumbv6m-none-eabi",
+                    "--features",
+                    "print-defmt",
+                ],
+                Some("firmware/panic-probe"),
+                &[],
+            )
+        },
+        "cross",
+        errors,
+    );
+
+    do_test(
+        || {
+            run_command::<&str>(
+                &[
+                    "cargo",
+                    "check",
+                    "--target",
+                    "thumbv6m-none-eabi",
+                    "--features",
+                    "print-rtt",
+                ],
+                Some("firmware/panic-probe"),
+                &[],
+            )
+        },
+        "cross",
+        errors,
+    )
+}
+
+fn test_snapshot(errors: &mut Vec<String>) {
+    println!("🧪 qemu/snapshot");
+    let mut tests = vec![
+        "log",
+        "timestamp",
+        "panic",
+        "assert",
+        "assert-eq",
+        "assert-ne",
+        "unwrap",
+        "defmt-test",
+        "hints",
+    ];
+
+    match rustc_is_nightly() {
+        Ok(is_nightly) => {
+            if is_nightly {
+                tests.push("alloc");
+            }
+        }
+        Err(e) => {
+            eprintln!(
+                "could not determine whether rust compiler is nightly - assuming it's not ({})",
```

Comment:
You can unwrap this result – if we cannot run `rustc` something is very wrong

Reply:
aye, I wasn't sure about this one either. Mainly kept my approach to *always* be able to uninstall the previously installed targets. But it's such a strange edge case that people will probably not complain about lingering targets ever.